library/Theme/CustomModuleHelper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Simrishamn\Theme;
+
+class CustomModuleHelper
+{
+    
+    const DOMAIN        = 'simrishamn';
+    const MODULE_FOLDER = SIMRISHAMN_PATH . '/custom-modules/';
+    
+    /**
+     * Initialize the Module
+     *
+     * @param object $object Current Class object
+     * @return object The module settings
+     */
+    public static function setModule ($object) : object
+    {
+        
+        if (!$object)
+            throw new \Exception('No object was found.');
+        
+        $obj = new \stdClass();
+        
+        $obj->moduleName    = self::moduleName($object);
+        
+        $obj->slug          = self::kebabcase($obj->moduleName);
+        $obj->fields        = self::fields($obj->moduleName);
+        
+        $obj->nameSingular  = self::translate($object->nameSingular);
+        $obj->namePlural    = self::translate($object->namePlural);
+        $obj->description   = self::translate($object->description);
+        
+        return $obj;
+        
+    }
+    
+    /**
+     * Current Class name
+     *
+     * @param string $object Current Class object
+     * @return string The module name
+     */
+    public static function moduleName ($object) : string
+    {
+        
+        return (new \ReflectionClass($object))->getShortName();
+        
+    }
+    
+    /**
+     * Creates a slugified kebabcase string
+     *
+     * @param string $string The string to be converted
+     * @return string The converted string
+     */
+    public static function kebabcase ($string) : string
+    {
+        
+        $string = strip_tags($string);
+        
+        $string = preg_replace('~[^\pL\d]+~u', '-', $string);
+        
+        setlocale(LC_ALL, 'sv_SE.utf8');
+        
+        $string = iconv('utf-8', 'us-ascii//TRANSLIT', $string);
+
+        $string = preg_replace('~[^-\w]+~', '', $string);
+
+        $string = trim($string, '-');
+
+        $string = preg_replace('~-+~', '-', $string);
+        
+        $string = preg_replace(['/([a-z\d])([A-Z])/', '/([^-])([A-Z][a-z])/'], '$1-$2', $string);
+        
+        $string = strtolower($string);
+
+        if (empty($string))
+            throw new \Exception('No slug was set.');
+        
+        return $string;
+        
+    }
+    
+    /**
+     * Translate via WordPress __() function
+     *
+     * @param string $string The string to be translated
+     * @return string The translated string
+     */
+    public static function translate ($string) : string
+    {
+        
+        return __($string, self::DOMAIN);
```

Comment:
I don't think this will actually work when we regenerate the translations.

The reason is that poedit/gettext looks for `__("...", "...");` (amongst others) and takes whatever is in the first argument as the translatable string. That is, it doesn't actually run the PHP code to figure out which strings are available. Instead, it statically analyses the code and _might_ not find the strings unless they're right there in the argument.

We'll have to verify this first!